docs/iris/src/further_topics/metadata.rst

```diff
@@ -0,0 +1,987 @@
+
+Akin to the behaviour of a `namedtuple`_, the metadata classes in
+:numref:`metadata classes table` create **tuple-like** instances i.e., they provide a
+**snapshot** of the associated metadata member **values**, which are **not
+settable**, but they **may be mutable** depending on the data-type of the member.
+For example, given the following ``metadata`` of a :class:`~iris.coords.DimCoord`,
+
+    >>> longitude = cube.coord("longitude")
+    >>> metadata = longitude.metadata
+    >>> metadata
+    DimCoordMetadata(standard_name='longitude', long_name=None, var_name='longitude', units=Unit('degrees'), attributes={}, coord_system=GeogCS(6371229.0), climatological=False, circular=False)
+
+The ``metadata`` member value **is** the same as the container class member value,
+
+    >>> metadata.attributes is longitude.attributes
+    True
+    >>> metadata.circular is longitude.circular
+    True
+
+Like a `namedtuple`_, the ``metadata`` member is **not settable**,
+
+    >>> metadata.attributes = {"grinning face": "🙂"}
+    Traceback (most recent call last):
+    AttributeError: can't set attribute
+
+However, for a `dict`_ member, it **is mutable**,
+
+    >>> metadata.attributes
+    {}
+    >>> longitude.attributes["grinning face"] = "🙂"
+    >>> metadata.attributes
+    {'grinning face': '🙂'}
+    >>> metadata.attributes["grinning face"] = "🙃"
+    >>> longitude.attributes
+    {'grinning face': '🙃'}
+
+But ``metadata`` members with simple values are **not** mutable,
+
+    >>> metadata.circular
+    False
+    >>> longitude.circular = True
+    >>> metadata.circular
+    False
+
+And of course, they're also **not** settable,
+
+    >>> metadata.circular = True
+    Traceback (most recent call last):
+    AttributeError: can't set attribute
+
+Note that, the ``metadata`` property re-creates a **new** instance per invocation,
+with a **snapshot** of the container class metadata values at that point in time,
+
+    >>> longitude.metadata
+    DimCoordMetadata(standard_name='longitude', long_name=None, var_name='longitude', units=Unit('degrees'), attributes={'grinning face': '🙃'}, coord_system=GeogCS(6371229.0), climatological=False, circular=True)
+
+
+Metadata class behaviour
+------------------------
+
+As mentioned previously, the metadata classes in :numref:`metadata classes table`
+inherit the behaviour of a `namedtuple`_, and so act and feel like a `namedtuple`_,
+just as you might expect. For example, given the following ``metadata``,
+
+    >>> metadata
+    DimCoordMetadata(standard_name='longitude', long_name=None, var_name='longitude', units=Unit('degrees'), attributes={'grinning face': '🙃'}, coord_system=GeogCS(6371229.0), climatological=False, circular=False)
+
+We can use the `namedtuple._make`_ method to create a **new**
+:class:`~iris.common.metadata.DimCoordMetadata` instance from an existing sequence
+or iterable:
+
+    >>> values = (1, 2, 3, 4, 5, 6, 7, 8)
```

Comment:
Q: How do I know which metadata member the values are assigned to?  I'm assuming you have a secret hidden order?

Reply:
@abooton It's no secret, it's implied that it matches the `namedtuple._fields`. I've added clarification.